### PR TITLE
[Issue #6843] Replace concatenated strings, %-strings, and str.format with f-strings

### PR DESCRIPTION
**Pull request by sarahsmi**
Replace concatenated strings, %-strings, and str.format with f-strings

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [ ] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [ ] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/settings.py [hunk 1]
```diff
@@ -99,13 +99,11 @@
...
 if not SITEURL.endswith('/'):
-    SITEURL = '{}/'.format(SITEURL)
+    SITEURL = f'{SITEURL}/')
 
 DATABASE_URL = os.getenv(
     'DATABASE_URL',
-    'spatialite:///{path}'.format(
-        path=os.path.join(PROJECT_ROOT, 'development.db')
-    )
+    'spatialite:///{os.path.join(PROJECT_ROOT, 'development.db'}')
```
Comment (mtnorthcott) [inline geonode/settings.py:106]: Single quote problem

File: geonode/settings.py [hunk 2]
```diff
@@ -923,8 +921,8 @@
 # per-deployment settings should go here
 
 # Login and logout urls override
-LOGIN_URL = os.getenv('LOGIN_URL', '{}account/login/'.format(SITEURL))
-LOGOUT_URL = os.getenv('LOGOUT_URL', '{}account/logout/'.format(SITEURL))
+LOGIN_URL = os.getenv(f'LOGIN_URL', '{SITEURL}account/login/')
+LOGOUT_URL = os.getenv(f'LOGOUT_URL', '{SITEURL}account/logout/')
```
Comment (mtnorthcott) [inline geonode/settings.py:925]: `f` character is preceding the incorrect string

File: geonode/tests/smoke.py [hunk 3]
```diff
@@ -332,7 +332,7 @@ def test_inbox_redirects_when_not_logged_in(self):
         response = self.client.get(target_url)
         self.assertRedirects(
             response,
-            "{}{}?next=http%3A//testserver{}".format(settings.SITEURL[:-1], reverse("account_login"), target_url)
+            f"{settings.SITEURL[:-1]}{reverse("account_login")}?next=http%3A//testserver{target_url}"
```
Comment (mtnorthcott) [inline geonode/tests/smoke.py:335]: Double quote problem

File: geonode/tests/smoke.py [hunk 4]
```diff
@@ -353,7 +353,7 @@ def test_new_message_redirects_when_not_logged_in(self):
...
```
Comment (mtnorthcott) [inline geonode/tests/smoke.py:356]: Double quote problem

File: geonode/tests/smoke.py [hunk 5]
```diff
@@ -374,5 +374,5 @@ def test_thread_detail_redirects_when_not_logged_in(self):
...
```
Comment (mtnorthcott) [inline geonode/tests/smoke.py:377]: Double quote problem

File: geonode/upload/views.py [hunk 6]
```diff
@@ -165,24 +165,24 @@ def save_step_view(req, session):
     form = LayerUploadForm(req.POST, req.FILES)
     if form.is_valid():
         tempdir = tempfile.mkdtemp(dir=settings.FILE_UPLOAD_TEMP_DIR)
-        logger.debug("valid_extensions: {}".format(form.cleaned_data["valid_extensions"]))
+        logger.debug(f"valid_extensions: {form.cleaned_data["valid_extensions"]}")
```
Comment (mtnorthcott) [inline geonode/upload/views.py:168]: Double quote problem

File: geonode/urls.py [hunk 7]
```diff
@@ -234,7 +234,7 @@
     ]
 
 if settings.NOTIFICATIONS_MODULE in settings.INSTALLED_APPS:
-    notifications_urls = '{}.urls'.format(settings.NOTIFICATIONS_MODULE)
+    notifications_urls = fS'{settings.NOTIFICATIONS_MODULE}.urls'
```
Comment (mtnorthcott) [inline geonode/urls.py:237]: Typo

File: geonode/utils.py [hunk 8]
```diff
@@ -1086,10 +1084,7 @@ def build_caveats(resourcebase):
 
 
 def build_social_links(request, resourcebase):
-    social_url = "{protocol}://{host}{path}".format(
-        protocol=("https" if request.is_secure() else "http"),
-        host=request.get_host(),
-        path=request.get_full_path())
+    social_url = f"{("https" if request.is_secure() else "http")}://{request.get_host()}{request.get_full_path()}"
```
Comment (mtnorthcott) [inline geonode/utils.py:1087]: Double quote problem

File: geonode/utils.py [hunk 9]
```diff
@@ -1209,9 +1204,9 @@ def fixup_shp_columnnames(inShapefile, charset, tempdir=None):
                 while new_field_name in list_col_original or new_field_name in list_col.values():
                     if j == 0:
                         new_field_name += '_0'
-                    if new_field_name.endswith('_' + str(j)):
+                    if new_field_name.endswith(f'_{j}'):
                         j += 1
-                        new_field_name = new_field_name[:-2] + '_' + str(j)
+                        new_field_name = new_field_name[:-2] + f'_{j}'
```
Comment (mtnorthcott) [inline geonode/utils.py:1209]: Can also remove the concatenation with `f'{new_field_name[:-2]}_{j}'`

File: tasks.py [hunk 10]
```diff
@@ -553,7 +525,7 @@ def _prepare_monitoring_fixture():
         {
             "fields": {
                 "name": "geoserver-hostgeoserver",
-                "url": "{0}".format(os.environ['GEOSERVER_PUBLIC_LOCATION']),
+                "url": f"{os.environ['GEOSERVER_PUBLIC_LOCATION']S}",
```
Comment (mtnorthcott) [inline tasks.py:528]: Typo

**Review by mtnorthcott — CHANGES_REQUESTED** [review]
I probably haven't caught all occurrences but please check places where you have tried accessing dictionary keys within your curly braces. In some cases, the string is closed early which will result in a syntax error.

**Comment by afabiani** [comment]
Superseded by https://github.com/GeoNode/geonode/pull/7096

Please use that branch if you need to continue working on that one.

**Pull request closed without merging**